// @fluentui/react-infobutton/lib-commonjs/components/InfoLabel/useInfoLabel.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useInfoLabel_unstable", {
    enumerable: true,
    get: ()=>useInfoLabel_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactLabel = require("@fluentui/react-label");
const _reactUtilities = require("@fluentui/react-utilities");
const _infoButton = require("../InfoButton/InfoButton");
var _infoButton1, _arialabelledby, _root, _ariaowns;
const useInfoLabel_unstable = (props, ref)=>{
    const { root: rootShorthand , label: labelShorthand , infoButton: infoButtonShorthand , info , size , className , style , ...labelProps } = props;
    const baseId = (0, _reactUtilities.useId)('infolabel-');
    const [open, setOpen] = _react.useState(false);
    const root = (0, _reactUtilities.resolveShorthand)(rootShorthand, {
        required: true,
        defaultProps: {
            className,
            style
        }
    });
    const label = (0, _reactUtilities.resolveShorthand)(labelShorthand, {
        required: true,
        defaultProps: {
            id: baseId + '__label',
            ref,
            size,
            ...labelProps
        }
    });
    const infoButton = (0, _reactUtilities.resolveShorthand)(infoButtonShorthand, {
        required: !!info,
        defaultProps: {
            id: baseId + '__infoButton',
            size,
            info
        }
    });
    const infoButtonPopover = (0, _reactUtilities.resolveShorthand)(infoButton === null || infoButton === void 0 ? void 0 : infoButton.popover, {
        required: true
    });
    infoButtonPopover.onOpenChange = (0, _reactUtilities.useEventCallback)((0, _reactUtilities.mergeCallbacks)(infoButtonPopover.onOpenChange, (e, data)=>{
        setOpen(data.open);
    }));
    if (infoButton) {
        infoButton.popover = infoButtonPopover;
        infoButton.info = (0, _reactUtilities.resolveShorthand)(infoButton === null || infoButton === void 0 ? void 0 : infoButton.info, {
            defaultProps: {
                id: baseId + '__info'
            }
        });
        var _;
        (_ = (_infoButton1 = infoButton)[_arialabelledby = 'aria-labelledby']) !== null && _ !== void 0 ? _ : _infoButton1[_arialabelledby] = `${label.id} ${infoButton.id}`;
        if (open) {
            var _infoButton_info;
            var _1;
            (_1 = (_root = root)[_ariaowns = 'aria-owns']) !== null && _1 !== void 0 ? _1 : _root[_ariaowns] = (_infoButton_info = infoButton.info) === null || _infoButton_info === void 0 ? void 0 : _infoButton_info.id;
        }
    }
    return {
        size,
        components: {
            root: 'span',
            label: _reactLabel.Label,
            infoButton: _infoButton.InfoButton
        },
        root,
        label,
        infoButton
    };
};
