// @fluentui/react-drawer/lib-commonjs/components/DrawerBody/useDrawerBodyStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    drawerBodyClassNames: ()=>drawerBodyClassNames,
    useDrawerBodyStyles_unstable: ()=>useDrawerBodyStyles_unstable
});
const _react = require("@griffel/react");
const drawerBodyClassNames = {
    root: 'fui-DrawerBody'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        B6of3ja: "fkrn0sh",
        t21cq0: [
            "f11qmguv",
            "f1tyq0we"
        ],
        jrapky: "fmxx68s",
        Frg6f3: [
            "f1tyq0we",
            "f11qmguv"
        ],
        z8tnut: "f1sl3k7w",
        z189sj: [
            "fnbas0t",
            "f1mswpri"
        ],
        Byoj8tv: "f1brlhvm",
        uwmqm3: [
            "f1mswpri",
            "fnbas0t"
        ],
        B68tc82: "f1oy3dpc",
        Bmxbyg5: "f5zp4f",
        Bh6795r: "fqerorx",
        Bnnss6s: "f1neuvcm",
        xawz: "f1pztt34",
        qb2dma: "f1ub7u0d",
        Bcmaq0h: "f1igvcbx",
        B676am1: "f1fofb6i",
        e5kdtc: "fo6xqva",
        De3pzq: "fxugw4r",
        Bkjc3bi: "fcre6b6",
        Bgh53k4: "f1besv5s",
        Bcx2cba: "f1pqm9hp",
        gonrqn: "f2c46kh"
    }
}, {
    d: [
        ".fkrn0sh{margin-top:-1px;}",
        ".f11qmguv{margin-right:0;}",
        ".f1tyq0we{margin-left:0;}",
        ".fmxx68s{margin-bottom:-1px;}",
        ".f1sl3k7w{padding-top:1px;}",
        ".fnbas0t{padding-right:var(--spacingHorizontalXXL);}",
        ".f1mswpri{padding-left:var(--spacingHorizontalXXL);}",
        ".f1brlhvm{padding-bottom:1px;}",
        ".f1oy3dpc{overflow-x:auto;}",
        ".f5zp4f{overflow-y:auto;}",
        ".fqerorx{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}",
        ".f1neuvcm{-webkit-flex-shrink:1;-ms-flex-negative:1;flex-shrink:1;}",
        ".f1pztt34{-webkit-flex-basis:0;-ms-flex-preferred-size:0;flex-basis:0;}",
        ".f1ub7u0d{-webkit-align-self:stretch;-ms-flex-item-align:stretch;-ms-grid-row-align:stretch;align-self:stretch;}",
        ".f1igvcbx{background-image:linear-gradient(to top, var(--colorNeutralBackground1), var(--colorNeutralBackground1)),linear-gradient(to top, var(--colorNeutralBackground1), var(--colorNeutralBackground1)),linear-gradient(to top, var(--colorNeutralStroke1), var(--colorNeutralBackground1)),linear-gradient(to bottom, var(--colorNeutralStroke1), var(--colorNeutralBackground1));}",
        ".f1fofb6i{-webkit-background-position:bottom center,top center,bottom center,top center;background-position:bottom center,top center,bottom center,top center;}",
        ".fo6xqva{background-repeat:no-repeat;}",
        ".fxugw4r{background-color:var(--colorNeutralBackground1);}",
        ".fcre6b6{-webkit-background-size:100% 2px,100% 2px,100% 1px,100% 1px;background-size:100% 2px,100% 2px,100% 1px,100% 1px;}",
        ".f1besv5s{background-attachment:local,local,scroll,scroll;}",
        ".f1pqm9hp:last-child{padding-bottom:calc(var(--spacingHorizontalXXL) + 1px);}",
        ".f2c46kh:first-child{padding-top:calc(var(--spacingHorizontalXXL) + 1px);}"
    ]
});
const useDrawerBodyStyles_unstable = (state)=>{
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(drawerBodyClassNames.root, styles.root, state.root.className);
    return state;
}; //# sourceMappingURL=useDrawerBodyStyles.styles.js.map
