// @fluentui/react-card/lib-commonjs/components/CardHeader/useCardHeader.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useCardHeader_unstable", {
    enumerable: true,
    get: ()=>useCardHeader_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _cardContext = require("../Card/CardContext");
const _useCardHeaderStylesStyles = require("./useCardHeaderStyles.styles");
/**
 * Finds the first child of CardHeader with an id prop.
 *
 * @param header - the header prop of CardHeader
 */ function getChildWithId(header) {
    function isReactElementWithIdProp(element) {
        return /*#__PURE__*/ _react.isValidElement(element) && Boolean(element.props.id);
    }
    return _react.Children.toArray(header).find(isReactElementWithIdProp);
}
/**
 * Returns the id to use for the CardHeader root element.
 *
 * @param headerId - the id prop of the CardHeader component
 * @param childWithId - the first child of the CardHeader component with an id prop
 * @param generatedId - a generated id
 *
 * @returns the id to use for the CardHeader root element
 */ function getReferenceId(headerId, childWithId, generatedId) {
    if (headerId) {
        return headerId;
    }
    if (childWithId === null || childWithId === void 0 ? void 0 : childWithId.props.id) {
        return childWithId.props.id;
    }
    return generatedId;
}
const useCardHeader_unstable = (props, ref)=>{
    const { image , header , description , action  } = props;
    const { selectableA11yProps: { referenceId , setReferenceId  }  } = (0, _cardContext.useCardContext_unstable)();
    const headerRef = _react.useRef(null);
    const hasChildId = _react.useRef(false);
    const generatedId = (0, _reactUtilities.useId)(_useCardHeaderStylesStyles.cardHeaderClassNames.header, referenceId);
    const headerSlot = (0, _reactUtilities.resolveShorthand)(header, {
        required: true,
        defaultProps: {
            ref: headerRef,
            id: !hasChildId.current ? referenceId : undefined
        }
    });
    _react.useEffect(()=>{
        var _headerRef_current;
        const headerId = !hasChildId.current ? (_headerRef_current = headerRef.current) === null || _headerRef_current === void 0 ? void 0 : _headerRef_current.id : undefined;
        const childWithId = getChildWithId(headerSlot === null || headerSlot === void 0 ? void 0 : headerSlot.children);
        hasChildId.current = Boolean(childWithId);
        setReferenceId(getReferenceId(headerId, childWithId, generatedId));
    }, [
        generatedId,
        header,
        headerSlot,
        setReferenceId
    ]);
    return {
        components: {
            root: 'div',
            image: 'div',
            header: 'div',
            description: 'div',
            action: 'div'
        },
        root: (0, _reactUtilities.getNativeElementProps)('div', {
            ref,
            ...props
        }),
        image: (0, _reactUtilities.resolveShorthand)(image),
        header: headerSlot,
        description: (0, _reactUtilities.resolveShorthand)(description),
        action: (0, _reactUtilities.resolveShorthand)(action)
    };
};
