// @fluentui/react-virtualizer/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Virtualizer: ()=>_virtualizer.Virtualizer,
    virtualizerClassNames: ()=>_virtualizer.virtualizerClassNames,
    useVirtualizer_unstable: ()=>_virtualizer.useVirtualizer_unstable,
    renderVirtualizer_unstable: ()=>_virtualizer.renderVirtualizer_unstable,
    useVirtualizerStyles_unstable: ()=>_virtualizer.useVirtualizerStyles_unstable,
    useIntersectionObserver: ()=>_hooks.useIntersectionObserver,
    useStaticVirtualizerMeasure: ()=>_hooks.useStaticVirtualizerMeasure,
    useDynamicVirtualizerMeasure: ()=>_hooks.useDynamicVirtualizerMeasure,
    useResizeObserverRef_unstable: ()=>_hooks.useResizeObserverRef_unstable,
    VirtualizerContextProvider: ()=>_utilities.VirtualizerContextProvider,
    useVirtualizerContext_unstable: ()=>_utilities.useVirtualizerContext_unstable,
    scrollToItemStatic: ()=>_utilities.scrollToItemStatic,
    scrollToItemDynamic: ()=>_utilities.scrollToItemDynamic,
    VirtualizerScrollView: ()=>_virtualizerScrollView.VirtualizerScrollView,
    virtualizerScrollViewClassNames: ()=>_virtualizerScrollView.virtualizerScrollViewClassNames,
    useVirtualizerScrollView_unstable: ()=>_virtualizerScrollView.useVirtualizerScrollView_unstable,
    renderVirtualizerScrollView_unstable: ()=>_virtualizerScrollView.renderVirtualizerScrollView_unstable,
    useVirtualizerScrollViewStyles_unstable: ()=>_virtualizerScrollView.useVirtualizerScrollViewStyles_unstable,
    VirtualizerScrollViewDynamic: ()=>_virtualizerScrollViewDynamic.VirtualizerScrollViewDynamic,
    virtualizerScrollViewDynamicClassNames: ()=>_virtualizerScrollViewDynamic.virtualizerScrollViewDynamicClassNames,
    useVirtualizerScrollViewDynamic_unstable: ()=>_virtualizerScrollViewDynamic.useVirtualizerScrollViewDynamic_unstable,
    renderVirtualizerScrollViewDynamic_unstable: ()=>_virtualizerScrollViewDynamic.renderVirtualizerScrollViewDynamic_unstable,
    useVirtualizerScrollViewDynamicStyles_unstable: ()=>_virtualizerScrollViewDynamic.useVirtualizerScrollViewDynamicStyles_unstable
});
const _virtualizer = require("./Virtualizer");
const _hooks = require("./Hooks");
const _utilities = require("./Utilities");
const _virtualizerScrollView = require("./VirtualizerScrollView");
const _virtualizerScrollViewDynamic = require("./VirtualizerScrollViewDynamic");
