// @fluentui/react-tree/lib/components/TreeItemPersonaLayout/useTreeItemPersonaLayoutStyles.styles.js
import { __styles, mergeClasses, shorthands } from '@griffel/react';
import { tokens, typographyStyles } from '@fluentui/react-theme';
import { treeItemLevelToken } from '../../utils/tokens';
import { useTreeItemContext_unstable } from '../../contexts/treeItemContext';
export const treeItemPersonaLayoutClassNames = {
  root: 'fui-TreeItemPersonaLayout',
  media: 'fui-TreeItemPersonaLayout__media',
  description: 'fui-TreeItemPersonaLayout__description',
  content: 'fui-TreeItemPersonaLayout__content',
  expandIcon: 'fui-TreeItemPersonaLayout__expandIcon',
  aside: 'fui-TreeItemPersonaLayout__aside',
  actions: 'fui-TreeItemPersonaLayout__actions',
  selector: 'fui-TreeItemPersonaLayout__selector'
};
/**
 * Styles for the root slot
 */
const useRootStyles = /*#__PURE__*/__styles({
  base: {
    mc9l5x: "f13qh94s",
    wkccdc: "f1y0tuzo",
    Budl1dq: "fellwe7",
    zoa1oz: "f86xsft",
    Bt984gj: "f122n59",
    Bahqtrf: "fk6fouc",
    Be2twd7: "fkhj508",
    Bhrd7zp: "figsok6",
    Bg96gwp: "f1i3iumi",
    lj723h: "flvvhsy",
    ecr2s2: "f1wfn5kd",
    zt0xaq: "f5na5aj",
    Bi91k9c: "fnwyq0v",
    Jwef8y: "f1t94bn6",
    tbva4x: "f1oboesa"
  },
  leaf: {
    uwmqm3: ["f1k1erfc", "faevyjx"]
  },
  branch: {
    uwmqm3: ["fo100m9", "f6yw3pu"]
  }
}, {
  d: [".f13qh94s{display:-ms-grid;display:grid;}", ".f1y0tuzo{-ms-grid-rows:1fr auto;grid-template-rows:1fr auto;}", ".fellwe7{-ms-grid-columns:auto auto 1fr auto;grid-template-columns:auto auto 1fr auto;}", ".f86xsft{grid-template-areas:\"expandIcon media content        aside\" \"expandIcon media description aside\";}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".fk6fouc{font-family:var(--fontFamilyBase);}", ".fkhj508{font-size:var(--fontSizeBase300);}", ".figsok6{font-weight:var(--fontWeightRegular);}", ".f1i3iumi{line-height:var(--lineHeightBase300);}", ".f1k1erfc{padding-left:calc(var(--fluent-TreeItem--level, 1) * var(--spacingHorizontalXXL));}", ".faevyjx{padding-right:calc(var(--fluent-TreeItem--level, 1) * var(--spacingHorizontalXXL));}", ".fo100m9{padding-left:calc((var(--fluent-TreeItem--level, 1) - 1) * var(--spacingHorizontalXXL));}", ".f6yw3pu{padding-right:calc((var(--fluent-TreeItem--level, 1) - 1) * var(--spacingHorizontalXXL));}"],
  a: [".flvvhsy:active{color:var(--colorNeutralForeground2Pressed);}", ".f1wfn5kd:active{background-color:var(--colorSubtleBackgroundPressed);}", ".f5na5aj:active .fui-TreeItemPersonaLayout__expandIcon{color:var(--colorNeutralForeground3Pressed);}"],
  h: [".fnwyq0v:hover{color:var(--colorNeutralForeground2Hover);}", ".f1t94bn6:hover{background-color:var(--colorSubtleBackgroundHover);}", ".f1oboesa:hover .fui-TreeItemPersonaLayout__expandIcon{color:var(--colorNeutralForeground3Hover);}"]
});
/**
 * Styles for the expand icon slot
 */
const useMediaStyles = /*#__PURE__*/__styles({
  base: {
    mc9l5x: "f22iagw",
    Bt984gj: "f122n59",
    a9b677: "f1szoe96",
    Bqenvij: "f1d2rq10",
    Ijaq50: "f11uok23",
    Br312pm: "f1qdfnnj",
    nk6f5a: "f1s27gz",
    Bw0ie65: "f86d0yl",
    z8tnut: "f1g0x7ka",
    z189sj: ["f7x41pl", "fruq291"],
    Byoj8tv: "f1qch9an",
    uwmqm3: ["fgiv446", "ffczdla"]
  }
}, {
  d: [".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".f1szoe96{width:32px;}", ".f1d2rq10{height:32px;}", ".f11uok23{-ms-grid-row:media;grid-row-start:media;}", ".f1qdfnnj{-ms-grid-column:media;grid-column-start:media;}", ".f1s27gz{-ms-grid-row-span:media;grid-row-end:media;}", ".f86d0yl{-ms-grid-column-span:media;grid-column-end:media;}", ".f1g0x7ka{padding-top:0;}", ".f7x41pl{padding-right:var(--spacingHorizontalXS);}", ".fruq291{padding-left:var(--spacingHorizontalXS);}", ".f1qch9an{padding-bottom:0;}", ".fgiv446{padding-left:var(--spacingHorizontalXXS);}", ".ffczdla{padding-right:var(--spacingHorizontalXXS);}"]
});
const useContentStyles = /*#__PURE__*/__styles({
  base: {
    Ijaq50: "fxwmwwq",
    Br312pm: "f1e0gndy",
    nk6f5a: "f4zz6gv",
    Bw0ie65: "f7idnck",
    z8tnut: "f1ngh7ph",
    z189sj: ["f7x41pl", "fruq291"],
    Byoj8tv: "f5o476b",
    uwmqm3: ["f1f5gg8d", "f1vdfbxk"]
  },
  withDescription: {
    z8tnut: "f1ngh7ph",
    z189sj: ["f7x41pl", "fruq291"],
    Byoj8tv: "f1qch9an",
    uwmqm3: ["f1f5gg8d", "f1vdfbxk"]
  }
}, {
  d: [".fxwmwwq{-ms-grid-row:content;grid-row-start:content;}", ".f1e0gndy{-ms-grid-column:content;grid-column-start:content;}", ".f4zz6gv{-ms-grid-row-span:content;grid-row-end:content;}", ".f7idnck{-ms-grid-column-span:content;grid-column-end:content;}", ".f1ngh7ph{padding-top:var(--spacingVerticalMNudge);}", ".f7x41pl{padding-right:var(--spacingHorizontalXS);}", ".fruq291{padding-left:var(--spacingHorizontalXS);}", ".f5o476b{padding-bottom:var(--spacingVerticalMNudge);}", ".f1f5gg8d{padding-left:var(--spacingHorizontalS);}", ".f1vdfbxk{padding-right:var(--spacingHorizontalS);}", ".f1qch9an{padding-bottom:0;}"]
});
const useDescriptionStyles = /*#__PURE__*/__styles({
  base: {
    Ijaq50: "ffekjdo",
    Br312pm: "f12wtlaa",
    nk6f5a: "f1blqr63",
    Bw0ie65: "f2ve1i",
    Bahqtrf: "fk6fouc",
    Be2twd7: "fy9rknc",
    Bhrd7zp: "figsok6",
    Bg96gwp: "fwrc4pm",
    z8tnut: "f1g0x7ka",
    z189sj: ["f7x41pl", "fruq291"],
    Byoj8tv: "f5o476b",
    uwmqm3: ["f1f5gg8d", "f1vdfbxk"]
  }
}, {
  d: [".ffekjdo{-ms-grid-row:description;grid-row-start:description;}", ".f12wtlaa{-ms-grid-column:description;grid-column-start:description;}", ".f1blqr63{-ms-grid-row-span:description;grid-row-end:description;}", ".f2ve1i{-ms-grid-column-span:description;grid-column-end:description;}", ".fk6fouc{font-family:var(--fontFamilyBase);}", ".fy9rknc{font-size:var(--fontSizeBase200);}", ".figsok6{font-weight:var(--fontWeightRegular);}", ".fwrc4pm{line-height:var(--lineHeightBase200);}", ".f1g0x7ka{padding-top:0;}", ".f7x41pl{padding-right:var(--spacingHorizontalXS);}", ".fruq291{padding-left:var(--spacingHorizontalXS);}", ".f5o476b{padding-bottom:var(--spacingVerticalMNudge);}", ".f1f5gg8d{padding-left:var(--spacingHorizontalS);}", ".f1vdfbxk{padding-right:var(--spacingHorizontalS);}"]
});
/**
 * Styles for the action icon slot
 */
const useActionsStyles = /*#__PURE__*/__styles({
  base: {
    mc9l5x: "f22iagw",
    Frg6f3: ["fcgxt0o", "f1ujusj6"],
    qhf8xq: "f10pi13n",
    Bj3rh1h: "f19g0ac",
    Ijaq50: "fobksn0",
    Br312pm: "fmy5l6f",
    nk6f5a: "fzqypwc",
    Bw0ie65: "f1tmftl3",
    z8tnut: "f1g0x7ka",
    z189sj: ["f1vdfbxk", "f1f5gg8d"],
    Byoj8tv: "f1qch9an",
    uwmqm3: ["f1f5gg8d", "f1vdfbxk"]
  }
}, {
  d: [".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".fcgxt0o{margin-left:auto;}", ".f1ujusj6{margin-right:auto;}", ".f10pi13n{position:relative;}", ".f19g0ac{z-index:1;}", ".fobksn0{-ms-grid-row:aside;grid-row-start:aside;}", ".fmy5l6f{-ms-grid-column:aside;grid-column-start:aside;}", ".fzqypwc{-ms-grid-row-span:aside;grid-row-end:aside;}", ".f1tmftl3{-ms-grid-column-span:aside;grid-column-end:aside;}", ".f1g0x7ka{padding-top:0;}", ".f1vdfbxk{padding-right:var(--spacingHorizontalS);}", ".f1f5gg8d{padding-left:var(--spacingHorizontalS);}", ".f1qch9an{padding-bottom:0;}"]
});
/**
 * Styles for the action icon slot
 */
const useAsideStyles = /*#__PURE__*/__styles({
  base: {
    mc9l5x: "f22iagw",
    Frg6f3: ["fcgxt0o", "f1ujusj6"],
    Bt984gj: "f122n59",
    Bj3rh1h: "f11zp4z2",
    Ijaq50: "fobksn0",
    Br312pm: "fmy5l6f",
    nk6f5a: "fzqypwc",
    Bw0ie65: "f1tmftl3",
    z8tnut: "f1g0x7ka",
    z189sj: ["fw5db7e", "f1uw59to"],
    Byoj8tv: "f1qch9an",
    uwmqm3: ["f1uw59to", "fw5db7e"],
    i8kkvl: "f1ufnopg",
    Belr9w4: "f14sijuj"
  }
}, {
  d: [".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".fcgxt0o{margin-left:auto;}", ".f1ujusj6{margin-right:auto;}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".f11zp4z2{z-index:0;}", ".fobksn0{-ms-grid-row:aside;grid-row-start:aside;}", ".fmy5l6f{-ms-grid-column:aside;grid-column-start:aside;}", ".fzqypwc{-ms-grid-row-span:aside;grid-row-end:aside;}", ".f1tmftl3{-ms-grid-column-span:aside;grid-column-end:aside;}", ".f1g0x7ka{padding-top:0;}", ".fw5db7e{padding-right:var(--spacingHorizontalM);}", ".f1uw59to{padding-left:var(--spacingHorizontalM);}", ".f1qch9an{padding-bottom:0;}", ".f1ufnopg{-webkit-column-gap:var(--spacingHorizontalXS);column-gap:var(--spacingHorizontalXS);}", ".f14sijuj{row-gap:var(--spacingHorizontalXS);}"]
});
/**
 * Styles for the expand icon slot
 */
const useExpandIconStyles = /*#__PURE__*/__styles({
  base: {
    mc9l5x: "f22iagw",
    Bt984gj: "f122n59",
    Brf1p80: "f4d9j23",
    Bf4jedk: "f17fgpbq",
    B7ck84d: "f1ewtqcl",
    sj55zd: "f11d4kpn",
    Ijaq50: "f1e6rimv",
    Br312pm: "f10hsf66",
    nk6f5a: "foflfm0",
    Bw0ie65: "f1b3ebjb",
    Bh6795r: "f1jhi6b8",
    Bnnss6s: "fi64zpg",
    xawz: "f1rmlqtg",
    z8tnut: "f1ywm7hm",
    z189sj: ["fhxju0i", "f1cnd47f"],
    Byoj8tv: "f14wxoun",
    uwmqm3: ["f1cnd47f", "fhxju0i"]
  }
}, {
  d: [".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".f4d9j23{-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}", ".f17fgpbq{min-width:24px;}", ".f1ewtqcl{box-sizing:border-box;}", ".f11d4kpn{color:var(--colorNeutralForeground3);}", ".f1e6rimv{-ms-grid-row:expandIcon;grid-row-start:expandIcon;}", ".f10hsf66{-ms-grid-column:expandIcon;grid-column-start:expandIcon;}", ".foflfm0{-ms-grid-row-span:expandIcon;grid-row-end:expandIcon;}", ".f1b3ebjb{-ms-grid-column-span:expandIcon;grid-column-end:expandIcon;}", ".f1jhi6b8{-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;}", ".fi64zpg{-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;}", ".f1rmlqtg{-webkit-flex-basis:auto;-ms-flex-preferred-size:auto;flex-basis:auto;}", ".f1ywm7hm{padding-top:var(--spacingVerticalXS);}", ".fhxju0i{padding-right:0;}", ".f1cnd47f{padding-left:0;}", ".f14wxoun{padding-bottom:var(--spacingVerticalXS);}"]
});
/**
 * Apply styling to the TreeItemPersonaLayout slots based on the state
 */
export const useTreeItemPersonaLayoutStyles_unstable = state => {
  const rootStyles = useRootStyles();
  const mediaStyles = useMediaStyles();
  const descriptionStyles = useDescriptionStyles();
  const actionsStyles = useActionsStyles();
  const asideStyles = useAsideStyles();
  const expandIconStyles = useExpandIconStyles();
  const contentStyles = useContentStyles();
  const itemType = useTreeItemContext_unstable(ctx => ctx.itemType);
  state.root.className = mergeClasses(treeItemPersonaLayoutClassNames.root, rootStyles.base, rootStyles[itemType], state.root.className);
  state.media.className = mergeClasses(treeItemPersonaLayoutClassNames.media, mediaStyles.base, state.media.className);
  if (state.content) {
    state.content.className = mergeClasses(treeItemPersonaLayoutClassNames.content, contentStyles.base, state.description && contentStyles.withDescription, state.content.className);
  }
  if (state.description) {
    state.description.className = mergeClasses(treeItemPersonaLayoutClassNames.description, descriptionStyles.base, state.description.className);
  }
  if (state.actions) {
    state.actions.className = mergeClasses(treeItemPersonaLayoutClassNames.actions, actionsStyles.base, state.actions.className);
  }
  if (state.aside) {
    state.aside.className = mergeClasses(treeItemPersonaLayoutClassNames.aside, asideStyles.base, state.aside.className);
  }
  if (state.expandIcon) {
    state.expandIcon.className = mergeClasses(treeItemPersonaLayoutClassNames.expandIcon, expandIconStyles.base, state.expandIcon.className);
  }
  if (state.selector) {
    state.selector.className = mergeClasses(treeItemPersonaLayoutClassNames.selector, state.selector.className);
  }
  return state;
};
