// @fluentui/react-drawer/lib/components/DrawerInline/useDrawerInlineStyles.styles.js
import * as React from 'react';
import { __styles, mergeClasses, shorthands } from '@griffel/react';
import { useDrawerBaseStyles } from '../../util/useDrawerBaseStyles.styles';
import { tokens } from '@fluentui/react-theme';
export const drawerInlineClassNames = {
  root: 'fui-DrawerInline'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/__styles({
  root: {
    qhf8xq: "f10pi13n"
  },
  separatorLeft: {
    Bekrc4i: ["f1hqa2wf", "finvdd3"],
    vrafjx: ["fcdblym", "fjik90z"],
    h3c5rm: ["fa8zu9y", "f17e9lqh"]
  },
  separatorRight: {
    ibv6hh: ["finvdd3", "f1hqa2wf"],
    wvpqe5: ["fjik90z", "fcdblym"],
    zhjwy3: ["f17e9lqh", "fa8zu9y"]
  }
}, {
  d: [".f10pi13n{position:relative;}", ".f1hqa2wf{border-right-width:1px;}", ".finvdd3{border-left-width:1px;}", ".fcdblym{border-right-style:solid;}", ".fjik90z{border-left-style:solid;}", ".fa8zu9y{border-right-color:var(--colorNeutralBackground3);}", ".f17e9lqh{border-left-color:var(--colorNeutralBackground3);}"]
});
/**
 * Apply styling to the DrawerInline slots based on the state
 */
export const useDrawerInlineStyles_unstable = state => {
  const baseStyles = useDrawerBaseStyles();
  const styles = useStyles();
  const separatorClass = React.useMemo(() => {
    if (!state.separator) {
      return undefined;
    }
    return state.position === 'left' ? styles.separatorLeft : styles.separatorRight;
  }, [state.position, state.separator, styles.separatorRight, styles.separatorLeft]);
  state.root.className = mergeClasses(drawerInlineClassNames.root, baseStyles.root, styles.root, state.size && baseStyles[state.size], state.position && baseStyles[state.position], separatorClass, state.root.className);
  return state;
};
