// @fluentui/react-drawer/lib/components/DrawerFooter/useDrawerFooterStyles.styles.js
import { __styles, mergeClasses, shorthands } from '@griffel/react';
import { tokens } from '@fluentui/react-theme';
export const drawerFooterClassNames = {
  root: 'fui-DrawerFooter'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/__styles({
  root: {
    z8tnut: "fikn0iw",
    z189sj: ["fnbas0t", "f1mswpri"],
    Byoj8tv: "fz66xbh",
    uwmqm3: ["f1mswpri", "fnbas0t"],
    mc9l5x: "f22iagw",
    Brf1p80: "fbhxue7",
    Bt984gj: "f122n59",
    i8kkvl: "fsnqrgy"
  }
}, {
  d: [".fikn0iw{padding-top:var(--spacingVerticalL);}", ".fnbas0t{padding-right:var(--spacingHorizontalXXL);}", ".f1mswpri{padding-left:var(--spacingHorizontalXXL);}", ".fz66xbh{padding-bottom:var(--spacingVerticalXXL);}", ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".fbhxue7{-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".fsnqrgy{-webkit-column-gap:var(--spacingHorizontalS);column-gap:var(--spacingHorizontalS);}"]
});
/**
 * Apply styling to the DrawerFooter slots based on the state
 */
export const useDrawerFooterStyles_unstable = state => {
  const styles = useStyles();
  state.root.className = mergeClasses(drawerFooterClassNames.root, styles.root, state.root.className);
  return state;
};
