// @fluentui/react-combobox/lib/utils/useComboboxPopup.js
import { resolvePositioningShorthand, usePositioning } from '@fluentui/react-positioning';
import { useMergedRefs } from '@fluentui/react-utilities';
export function useComboboxPopup(props, triggerShorthand, listboxShorthand) {
    const { positioning  } = props;
    // Set a default set of fallback positions to try if the dropdown does not fit on screen
    const fallbackPositions = [
        'above',
        'after',
        'after-top',
        'before',
        'before-top'
    ];
    // popper options
    const popperOptions = {
        position: 'below',
        align: 'start',
        offset: {
            crossAxis: 0,
            mainAxis: 2
        },
        fallbackPositions,
        ...resolvePositioningShorthand(positioning)
    };
    const { targetRef , containerRef  } = usePositioning(popperOptions);
    const listboxRef = useMergedRefs(listboxShorthand === null || listboxShorthand === void 0 ? void 0 : listboxShorthand.ref, containerRef);
    const listbox = listboxShorthand && {
        ...listboxShorthand,
        ref: listboxRef
    };
    return [
        {
            ...triggerShorthand,
            ref: useMergedRefs(triggerShorthand === null || triggerShorthand === void 0 ? void 0 : triggerShorthand.ref, targetRef)
        },
        listbox
    ];
}
