// @fluentui/react-drawer/lib/components/DrawerOverlay/useDrawerOverlayStyles.styles.js
import { __styles, mergeClasses } from '@griffel/react';
import { useDrawerBaseStyles } from '../../util/useDrawerBaseStyles.styles';
import * as React from 'react';
export const drawerOverlayClassNames = {
  root: 'fui-DrawerOverlay',
  backdrop: 'fui-DrawerOverlay__backdrop'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/__styles({
  root: {
    qhf8xq: "f19dog8a",
    Bhzewxz: "f15twtuk",
    B5kzvoi: "f1yab3r1"
  }
}, {
  d: [".f19dog8a{position:fixed;}", ".f15twtuk{top:0;}", ".f1yab3r1{bottom:0;}"]
});
/**
 * Apply styling to the DrawerOverlay slots based on the state
 */
export const useDrawerOverlayStyles_unstable = state => {
  const baseStyles = useDrawerBaseStyles();
  const styles = useStyles();
  const backdrop = state.root.backdrop;
  state.root.className = mergeClasses(drawerOverlayClassNames.root, baseStyles.root, styles.root, state.size && baseStyles[state.size], state.position && baseStyles[state.position], state.root.className);
  if (backdrop) {
    backdrop.className = mergeClasses(drawerOverlayClassNames.backdrop, backdrop.className);
  }
  return state;
};
