// @fluentui/react-drawer/lib-commonjs/components/DrawerHeader/useDrawerHeaderStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    drawerHeaderClassNames: ()=>drawerHeaderClassNames,
    useDrawerHeaderStyles_unstable: ()=>useDrawerHeaderStyles_unstable
});
const _react = require("@griffel/react");
const drawerHeaderClassNames = {
    root: 'fui-DrawerHeader'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        z8tnut: "f1ggmyuv",
        z189sj: [
            "fnbas0t",
            "f1mswpri"
        ],
        Byoj8tv: "f5b47ha",
        uwmqm3: [
            "f1mswpri",
            "fnbas0t"
        ],
        i8kkvl: "fsnqrgy",
        Belr9w4: "fylz90v",
        qb2dma: "f1ub7u0d",
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62"
    }
}, {
    d: [
        ".f1ggmyuv{padding-top:var(--spacingVerticalXXL);}",
        ".fnbas0t{padding-right:var(--spacingHorizontalXXL);}",
        ".f1mswpri{padding-left:var(--spacingHorizontalXXL);}",
        ".f5b47ha{padding-bottom:var(--spacingVerticalS);}",
        ".fsnqrgy{-webkit-column-gap:var(--spacingHorizontalS);column-gap:var(--spacingHorizontalS);}",
        ".fylz90v{row-gap:var(--spacingHorizontalS);}",
        ".f1ub7u0d{-webkit-align-self:stretch;-ms-flex-item-align:stretch;-ms-grid-row-align:stretch;align-self:stretch;}",
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}"
    ]
});
const useDrawerHeaderStyles_unstable = (state)=>{
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(drawerHeaderClassNames.root, styles.root, state.root.className);
    return state;
}; //# sourceMappingURL=useDrawerHeaderStyles.styles.js.map
