// @fluentui/react-shared-contexts/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ThemeContext_unstable: ()=>_themeContext.ThemeContext,
    ThemeProvider_unstable: ()=>_themeContext.ThemeProvider,
    ThemeClassNameProvider_unstable: ()=>_themeClassNameContext.ThemeClassNameProvider,
    useThemeClassName_unstable: ()=>_themeClassNameContext.useThemeClassName,
    TooltipVisibilityProvider_unstable: ()=>_tooltipVisibilityContext.TooltipVisibilityProvider,
    useTooltipVisibility_unstable: ()=>_tooltipVisibilityContext.useTooltipVisibility,
    Provider_unstable: ()=>_providerContext.Provider,
    useFluent_unstable: ()=>_providerContext.useFluent,
    OverridesProvider_unstable: ()=>_overridesContext.OverridesProvider,
    useOverrides_unstable: ()=>_overridesContext.useOverrides,
    CustomStyleHooksContext_unstable: ()=>_customStyleHooksContext.CustomStyleHooksContext,
    CustomStyleHooksProvider_unstable: ()=>_customStyleHooksContext.CustomStyleHooksProvider,
    useCustomStyleHook_unstable: ()=>_customStyleHooksContext.useCustomStyleHook,
    BackgroundAppearanceProvider: ()=>_backgroundAppearanceContext.BackgroundAppearanceProvider,
    useBackgroundAppearance: ()=>_backgroundAppearanceContext.useBackgroundAppearance,
    PortalMountNodeProvider: ()=>_portalMountNodeContext.PortalMountNodeProvider,
    usePortalMountNode: ()=>_portalMountNodeContext.usePortalMountNode
});
const _themeContext = require("./ThemeContext");
const _themeClassNameContext = require("./ThemeClassNameContext");
const _tooltipVisibilityContext = require("./TooltipVisibilityContext");
const _providerContext = require("./ProviderContext");
const _overridesContext = require("./OverridesContext");
const _customStyleHooksContext = require("./CustomStyleHooksContext");
const _backgroundAppearanceContext = require("./BackgroundAppearanceContext");
const _portalMountNodeContext = require("./PortalMountNodeContext");
