// @fluentui/react-drawer/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Drawer: ()=>_drawer.Drawer,
    renderDrawer_unstable: ()=>_drawer.renderDrawer_unstable,
    useDrawer_unstable: ()=>_drawer.useDrawer_unstable,
    DrawerOverlay: ()=>_drawerOverlay.DrawerOverlay,
    drawerOverlayClassNames: ()=>_drawerOverlay.drawerOverlayClassNames,
    renderDrawerOverlay_unstable: ()=>_drawerOverlay.renderDrawerOverlay_unstable,
    useDrawerOverlayStyles_unstable: ()=>_drawerOverlay.useDrawerOverlayStyles_unstable,
    useDrawerOverlay_unstable: ()=>_drawerOverlay.useDrawerOverlay_unstable,
    DrawerInline: ()=>_drawerInline.DrawerInline,
    drawerInlineClassNames: ()=>_drawerInline.drawerInlineClassNames,
    renderDrawerInline_unstable: ()=>_drawerInline.renderDrawerInline_unstable,
    useDrawerInlineStyles_unstable: ()=>_drawerInline.useDrawerInlineStyles_unstable,
    useDrawerInline_unstable: ()=>_drawerInline.useDrawerInline_unstable,
    DrawerBody: ()=>_drawerBody.DrawerBody,
    drawerBodyClassNames: ()=>_drawerBody.drawerBodyClassNames,
    renderDrawerBody_unstable: ()=>_drawerBody.renderDrawerBody_unstable,
    useDrawerBodyStyles_unstable: ()=>_drawerBody.useDrawerBodyStyles_unstable,
    useDrawerBody_unstable: ()=>_drawerBody.useDrawerBody_unstable,
    DrawerHeader: ()=>_drawerHeader.DrawerHeader,
    drawerHeaderClassNames: ()=>_drawerHeader.drawerHeaderClassNames,
    renderDrawerHeader_unstable: ()=>_drawerHeader.renderDrawerHeader_unstable,
    useDrawerHeaderStyles_unstable: ()=>_drawerHeader.useDrawerHeaderStyles_unstable,
    useDrawerHeader_unstable: ()=>_drawerHeader.useDrawerHeader_unstable,
    DrawerHeaderTitle: ()=>_drawerHeaderTitle.DrawerHeaderTitle,
    drawerHeaderTitleClassNames: ()=>_drawerHeaderTitle.drawerHeaderTitleClassNames,
    renderDrawerHeaderTitle_unstable: ()=>_drawerHeaderTitle.renderDrawerHeaderTitle_unstable,
    useDrawerHeaderTitleStyles_unstable: ()=>_drawerHeaderTitle.useDrawerHeaderTitleStyles_unstable,
    useDrawerHeaderTitle_unstable: ()=>_drawerHeaderTitle.useDrawerHeaderTitle_unstable,
    DrawerHeaderNavigation: ()=>_drawerHeaderNavigation.DrawerHeaderNavigation,
    drawerHeaderNavigationClassNames: ()=>_drawerHeaderNavigation.drawerHeaderNavigationClassNames,
    renderDrawerHeaderNavigation_unstable: ()=>_drawerHeaderNavigation.renderDrawerHeaderNavigation_unstable,
    useDrawerHeaderNavigationStyles_unstable: ()=>_drawerHeaderNavigation.useDrawerHeaderNavigationStyles_unstable,
    useDrawerHeaderNavigation_unstable: ()=>_drawerHeaderNavigation.useDrawerHeaderNavigation_unstable,
    DrawerFooter: ()=>_drawerFooter.DrawerFooter,
    drawerFooterClassNames: ()=>_drawerFooter.drawerFooterClassNames,
    renderDrawerFooter_unstable: ()=>_drawerFooter.renderDrawerFooter_unstable,
    useDrawerFooterStyles_unstable: ()=>_drawerFooter.useDrawerFooterStyles_unstable,
    useDrawerFooter_unstable: ()=>_drawerFooter.useDrawerFooter_unstable
});
const _drawer = require("./Drawer");
const _drawerOverlay = require("./DrawerOverlay");
const _drawerInline = require("./DrawerInline");
const _drawerBody = require("./DrawerBody");
const _drawerHeader = require("./DrawerHeader");
const _drawerHeaderTitle = require("./DrawerHeaderTitle");
const _drawerHeaderNavigation = require("./DrawerHeaderNavigation");
const _drawerFooter = require("./DrawerFooter");
