// @fluentui/react-tree/lib-commonjs/components/FlatTree/useFlatTreeStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    flatTreeClassNames: ()=>flatTreeClassNames,
    useFlatTreeStyles_unstable: ()=>useFlatTreeStyles_unstable
});
const _react = require("@griffel/react");
const flatTreeClassNames = {
    root: 'fui-FlatTree'
};
const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62",
        Belr9w4: "f1j0q4x9"
    },
    subtree: {
        Bh6795r: "fqerorx",
        Bnnss6s: "f1neuvcm",
        xawz: "f1s4axba",
        Ijaq50: "f1na4k6z",
        Br312pm: "fwt6ga",
        nk6f5a: "fi45nfw",
        Bw0ie65: "f10ort2y"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".f1j0q4x9{row-gap:var(--spacingVerticalXXS);}",
        ".fqerorx{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}",
        ".f1neuvcm{-webkit-flex-shrink:1;-ms-flex-negative:1;flex-shrink:1;}",
        ".f1s4axba{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;}",
        ".f1na4k6z{-ms-grid-row:subtree;grid-row-start:subtree;}",
        ".fwt6ga{-ms-grid-column:subtree;grid-column-start:subtree;}",
        ".fi45nfw{-ms-grid-row-span:subtree;grid-row-end:subtree;}",
        ".f10ort2y{-ms-grid-column-span:subtree;grid-column-end:subtree;}"
    ]
});
const useFlatTreeStyles_unstable = (state)=>{
    const styles = useStyles();
    const isSubTree = state.level > 0;
    state.root.className = (0, _react.mergeClasses)(flatTreeClassNames.root, styles.root, isSubTree && styles.subtree, state.root.className);
    return state;
}; //# sourceMappingURL=useFlatTreeStyles.styles.js.map
