// @fluentui/react-dialog/lib-commonjs/components/DialogSurface/useDialogSurface.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useDialogSurface_unstable", {
    enumerable: true,
    get: ()=>useDialogSurface_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _contexts = require("../../contexts");
const _keyboardKeys = require("@fluentui/keyboard-keys");
const useDialogSurface_unstable = (props, ref)=>{
    const modalType = (0, _contexts.useDialogContext_unstable)((ctx)=>ctx.modalType);
    const modalAttributes = (0, _contexts.useDialogContext_unstable)((ctx)=>ctx.modalAttributes);
    const dialogRef = (0, _contexts.useDialogContext_unstable)((ctx)=>ctx.dialogRef);
    const open = (0, _contexts.useDialogContext_unstable)((ctx)=>ctx.open);
    const requestOpenChange = (0, _contexts.useDialogContext_unstable)((ctx)=>ctx.requestOpenChange);
    const dialogTitleID = (0, _contexts.useDialogContext_unstable)((ctx)=>ctx.dialogTitleId);
    const handledBackdropClick = (0, _reactUtilities.useEventCallback)((event)=>{
        if ((0, _reactUtilities.isResolvedShorthand)(props.backdrop)) {
            var _props_backdrop, _props_backdrop_onClick;
            (_props_backdrop_onClick = (_props_backdrop = props.backdrop).onClick) === null || _props_backdrop_onClick === void 0 ? void 0 : _props_backdrop_onClick.call(_props_backdrop, event);
        }
        if (modalType === 'modal' && !event.isDefaultPrevented()) {
            requestOpenChange({
                event,
                open: false,
                type: 'backdropClick'
            });
        }
    });
    const handleKeyDown = (0, _reactUtilities.useEventCallback)((event)=>{
        var _props_onKeyDown;
        (_props_onKeyDown = props.onKeyDown) === null || _props_onKeyDown === void 0 ? void 0 : _props_onKeyDown.call(props, event);
        if (event.key === _keyboardKeys.Escape && !event.isDefaultPrevented()) {
            requestOpenChange({
                event,
                open: false,
                type: 'escapeKeyDown'
            });
            // stop propagation to avoid conflicting with other elements that listen for `Escape`
            // e,g: nested Dialog, Popover, Menu and Tooltip
            event.stopPropagation();
        }
    });
    const backdrop = (0, _reactUtilities.resolveShorthand)(props.backdrop, {
        required: open && modalType !== 'non-modal',
        defaultProps: {
            'aria-hidden': 'true'
        }
    });
    if (backdrop) {
        backdrop.onClick = handledBackdropClick;
    }
    var _props_as;
    return {
        components: {
            backdrop: 'div',
            root: 'div'
        },
        backdrop,
        root: (0, _reactUtilities.getNativeElementProps)((_props_as = props.as) !== null && _props_as !== void 0 ? _props_as : 'div', {
            tabIndex: -1,
            'aria-modal': modalType !== 'non-modal',
            role: modalType === 'alert' ? 'alertdialog' : 'dialog',
            'aria-labelledby': props['aria-label'] ? undefined : dialogTitleID,
            ...props,
            ...modalAttributes,
            onKeyDown: handleKeyDown,
            ref: (0, _reactUtilities.useMergedRefs)(ref, dialogRef)
        })
    };
};
