// @fluentui/react-tree/lib-commonjs/contexts/treeItemContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TreeItemContext: ()=>TreeItemContext,
    TreeItemProvider: ()=>TreeItemProvider,
    useTreeItemContext_unstable: ()=>useTreeItemContext_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactContextSelector = require("@fluentui/react-context-selector");
const defaultContextValue = {
    value: '',
    selectionRef: /*#__PURE__*/ _react.createRef(),
    layoutRef: /*#__PURE__*/ _react.createRef(),
    subtreeRef: /*#__PURE__*/ _react.createRef(),
    actionsRef: /*#__PURE__*/ _react.createRef(),
    expandIconRef: /*#__PURE__*/ _react.createRef(),
    isActionsVisible: false,
    isAsideVisible: false,
    itemType: 'leaf',
    open: false
};
const TreeItemContext = (0, _reactContextSelector.createContext)(undefined);
const { Provider: TreeItemProvider  } = TreeItemContext;
const useTreeItemContext_unstable = (selector)=>(0, _reactContextSelector.useContextSelector)(TreeItemContext, (ctx = defaultContextValue)=>selector(ctx));
