// @fluentui/react-tree/lib-commonjs/contexts/treeContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TreeContext: ()=>TreeContext,
    TreeProvider: ()=>TreeProvider,
    useTreeContext_unstable: ()=>useTreeContext_unstable
});
const _reactContextSelector = require("@fluentui/react-context-selector");
const _immutableSet = require("../utils/ImmutableSet");
const _immutableMap = require("../utils/ImmutableMap");
const defaultContextValue = {
    level: 0,
    selectionMode: 'none',
    openItems: _immutableSet.ImmutableSet.empty,
    checkedItems: _immutableMap.ImmutableMap.empty,
    requestTreeResponse: noop,
    appearance: 'subtle',
    size: 'medium'
};
function noop() {
/* noop */ }
const TreeContext = (0, _reactContextSelector.createContext)(undefined);
const { Provider: TreeProvider  } = TreeContext;
const useTreeContext_unstable = (selector)=>(0, _reactContextSelector.useContextSelector)(TreeContext, (ctx = defaultContextValue)=>selector(ctx));
