// @fluentui/react-components/lib-commonjs/unstable/index.js
// Stub for unstable exports
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Alert: ()=>_reactAlert.Alert,
    alertClassNames: ()=>_reactAlert.alertClassNames,
    renderAlert_unstable: ()=>_reactAlert.renderAlert_unstable,
    useAlertStyles_unstable: ()=>_reactAlert.useAlertStyles_unstable,
    useAlert_unstable: ()=>_reactAlert.useAlert_unstable,
    InfoButton: ()=>_reactInfobutton.InfoButton,
    infoButtonClassNames: ()=>_reactInfobutton.infoButtonClassNames,
    useInfoButton_unstable: ()=>_reactInfobutton.useInfoButton_unstable,
    useInfoButtonStyles_unstable: ()=>_reactInfobutton.useInfoButtonStyles_unstable,
    renderInfoButton_unstable: ()=>_reactInfobutton.renderInfoButton_unstable,
    InfoLabel: ()=>_reactInfobutton.InfoLabel,
    infoLabelClassNames: ()=>_reactInfobutton.infoLabelClassNames,
    renderInfoLabel_unstable: ()=>_reactInfobutton.renderInfoLabel_unstable,
    useInfoLabel_unstable: ()=>_reactInfobutton.useInfoLabel_unstable,
    useInfoLabelStyles_unstable: ()=>_reactInfobutton.useInfoLabelStyles_unstable,
    Virtualizer: ()=>_reactVirtualizer.Virtualizer,
    virtualizerClassNames: ()=>_reactVirtualizer.virtualizerClassNames,
    useVirtualizer_unstable: ()=>_reactVirtualizer.useVirtualizer_unstable,
    renderVirtualizer_unstable: ()=>_reactVirtualizer.renderVirtualizer_unstable,
    useVirtualizerStyles_unstable: ()=>_reactVirtualizer.useVirtualizerStyles_unstable,
    useIntersectionObserver: ()=>_reactVirtualizer.useIntersectionObserver,
    useStaticVirtualizerMeasure: ()=>_reactVirtualizer.useStaticVirtualizerMeasure,
    useDynamicVirtualizerMeasure: ()=>_reactVirtualizer.useDynamicVirtualizerMeasure,
    useResizeObserverRef_unstable: ()=>_reactVirtualizer.useResizeObserverRef_unstable,
    VirtualizerContextProvider: ()=>_reactVirtualizer.VirtualizerContextProvider,
    useVirtualizerContext_unstable: ()=>_reactVirtualizer.useVirtualizerContext_unstable,
    VirtualizerScrollView: ()=>_reactVirtualizer.VirtualizerScrollView,
    virtualizerScrollViewClassNames: ()=>_reactVirtualizer.virtualizerScrollViewClassNames,
    useVirtualizerScrollView_unstable: ()=>_reactVirtualizer.useVirtualizerScrollView_unstable,
    renderVirtualizerScrollView_unstable: ()=>_reactVirtualizer.renderVirtualizerScrollView_unstable,
    useVirtualizerScrollViewStyles_unstable: ()=>_reactVirtualizer.useVirtualizerScrollViewStyles_unstable,
    VirtualizerScrollViewDynamic: ()=>_reactVirtualizer.VirtualizerScrollViewDynamic,
    virtualizerScrollViewDynamicClassNames: ()=>_reactVirtualizer.virtualizerScrollViewDynamicClassNames,
    useVirtualizerScrollViewDynamic_unstable: ()=>_reactVirtualizer.useVirtualizerScrollViewDynamic_unstable,
    renderVirtualizerScrollViewDynamic_unstable: ()=>_reactVirtualizer.renderVirtualizerScrollViewDynamic_unstable,
    useVirtualizerScrollViewDynamicStyles_unstable: ()=>_reactVirtualizer.useVirtualizerScrollViewDynamicStyles_unstable,
    scrollToItemDynamic: ()=>_reactVirtualizer.scrollToItemDynamic,
    scrollToItemStatic: ()=>_reactVirtualizer.scrollToItemStatic,
    Tree: ()=>_reactTree.Tree,
    TreeItem: ()=>_reactTree.TreeItem,
    FlatTree: ()=>_reactTree.FlatTree,
    TreeProvider: ()=>_reactTree.TreeProvider,
    TreeItemLayout: ()=>_reactTree.TreeItemLayout,
    TreeItemProvider: ()=>_reactTree.TreeItemProvider,
    TreeItemPersonaLayout: ()=>_reactTree.TreeItemPersonaLayout,
    useTreeContext_unstable: ()=>_reactTree.useTreeContext_unstable,
    useTreeContextValues_unstable: ()=>_reactTree.useTreeContextValues_unstable,
    useTreeItemContext_unstable: ()=>_reactTree.useTreeItemContext_unstable,
    useTreeItemContextValues_unstable: ()=>_reactTree.useTreeItemContextValues_unstable,
    useFlatTreeContextValues_unstable: ()=>_reactTree.useFlatTreeContextValues_unstable,
    useTree_unstable: ()=>_reactTree.useTree_unstable,
    useTreeItem_unstable: ()=>_reactTree.useTreeItem_unstable,
    useTreeItemLayout_unstable: ()=>_reactTree.useTreeItemLayout_unstable,
    useTreeItemPersonaLayout_unstable: ()=>_reactTree.useTreeItemPersonaLayout_unstable,
    renderTree_unstable: ()=>_reactTree.renderTree_unstable,
    renderTreeItem_unstable: ()=>_reactTree.renderTreeItem_unstable,
    renderTreeItemPersonaLayout_unstable: ()=>_reactTree.renderTreeItemPersonaLayout_unstable,
    renderTreeItemLayout_unstable: ()=>_reactTree.renderTreeItemLayout_unstable,
    useTreeStyles_unstable: ()=>_reactTree.useTreeStyles_unstable,
    useTreeItemStyles_unstable: ()=>_reactTree.useTreeItemStyles_unstable,
    useTreeItemPersonaLayoutStyles_unstable: ()=>_reactTree.useTreeItemPersonaLayoutStyles_unstable,
    useTreeItemLayoutStyles_unstable: ()=>_reactTree.useTreeItemLayoutStyles_unstable,
    treeItemPersonaLayoutClassNames: ()=>_reactTree.treeItemPersonaLayoutClassNames,
    treeItemLevelToken: ()=>_reactTree.treeItemLevelToken,
    treeItemLayoutClassNames: ()=>_reactTree.treeItemLayoutClassNames,
    treeItemClassNames: ()=>_reactTree.treeItemClassNames,
    treeClassNames: ()=>_reactTree.treeClassNames,
    flatTreeClassNames: ()=>_reactTree.flatTreeClassNames,
    useFlatTree_unstable: ()=>_reactTree.useFlatTree_unstable,
    useHeadlessFlatTree_unstable: ()=>_reactTree.useHeadlessFlatTree_unstable,
    useFlatTreeStyles_unstable: ()=>_reactTree.useFlatTreeStyles_unstable,
    renderFlatTree_unstable: ()=>_reactTree.renderFlatTree_unstable,
    flattenTree_unstable: ()=>_reactTree.flattenTree_unstable,
    Drawer: ()=>_reactDrawer.Drawer,
    renderDrawer_unstable: ()=>_reactDrawer.renderDrawer_unstable,
    useDrawer_unstable: ()=>_reactDrawer.useDrawer_unstable,
    DrawerOverlay: ()=>_reactDrawer.DrawerOverlay,
    drawerOverlayClassNames: ()=>_reactDrawer.drawerOverlayClassNames,
    renderDrawerOverlay_unstable: ()=>_reactDrawer.renderDrawerOverlay_unstable,
    useDrawerOverlayStyles_unstable: ()=>_reactDrawer.useDrawerOverlayStyles_unstable,
    useDrawerOverlay_unstable: ()=>_reactDrawer.useDrawerOverlay_unstable,
    DrawerInline: ()=>_reactDrawer.DrawerInline,
    drawerInlineClassNames: ()=>_reactDrawer.drawerInlineClassNames,
    renderDrawerInline_unstable: ()=>_reactDrawer.renderDrawerInline_unstable,
    useDrawerInlineStyles_unstable: ()=>_reactDrawer.useDrawerInlineStyles_unstable,
    useDrawerInline_unstable: ()=>_reactDrawer.useDrawerInline_unstable,
    DrawerBody: ()=>_reactDrawer.DrawerBody,
    drawerBodyClassNames: ()=>_reactDrawer.drawerBodyClassNames,
    renderDrawerBody_unstable: ()=>_reactDrawer.renderDrawerBody_unstable,
    useDrawerBodyStyles_unstable: ()=>_reactDrawer.useDrawerBodyStyles_unstable,
    useDrawerBody_unstable: ()=>_reactDrawer.useDrawerBody_unstable,
    DrawerHeader: ()=>_reactDrawer.DrawerHeader,
    drawerHeaderClassNames: ()=>_reactDrawer.drawerHeaderClassNames,
    renderDrawerHeader_unstable: ()=>_reactDrawer.renderDrawerHeader_unstable,
    useDrawerHeaderStyles_unstable: ()=>_reactDrawer.useDrawerHeaderStyles_unstable,
    useDrawerHeader_unstable: ()=>_reactDrawer.useDrawerHeader_unstable,
    DrawerHeaderTitle: ()=>_reactDrawer.DrawerHeaderTitle,
    drawerHeaderTitleClassNames: ()=>_reactDrawer.drawerHeaderTitleClassNames,
    renderDrawerHeaderTitle_unstable: ()=>_reactDrawer.renderDrawerHeaderTitle_unstable,
    useDrawerHeaderTitleStyles_unstable: ()=>_reactDrawer.useDrawerHeaderTitleStyles_unstable,
    useDrawerHeaderTitle_unstable: ()=>_reactDrawer.useDrawerHeaderTitle_unstable,
    DrawerHeaderNavigation: ()=>_reactDrawer.DrawerHeaderNavigation,
    drawerHeaderNavigationClassNames: ()=>_reactDrawer.drawerHeaderNavigationClassNames,
    renderDrawerHeaderNavigation_unstable: ()=>_reactDrawer.renderDrawerHeaderNavigation_unstable,
    useDrawerHeaderNavigationStyles_unstable: ()=>_reactDrawer.useDrawerHeaderNavigationStyles_unstable,
    useDrawerHeaderNavigation_unstable: ()=>_reactDrawer.useDrawerHeaderNavigation_unstable,
    DrawerFooter: ()=>_reactDrawer.DrawerFooter,
    drawerFooterClassNames: ()=>_reactDrawer.drawerFooterClassNames,
    renderDrawerFooter_unstable: ()=>_reactDrawer.renderDrawerFooter_unstable,
    useDrawerFooterStyles_unstable: ()=>_reactDrawer.useDrawerFooterStyles_unstable,
    useDrawerFooter_unstable: ()=>_reactDrawer.useDrawerFooter_unstable
});
const _reactAlert = require("@fluentui/react-alert");
const _reactInfobutton = require("@fluentui/react-infobutton");
const _reactVirtualizer = require("@fluentui/react-virtualizer");
const _reactTree = require("@fluentui/react-tree");
const _reactDrawer = require("@fluentui/react-drawer");
