// @fluentui/react-dialog/lib/components/DialogActions/useDialogActionsStyles.styles.js
import { __styles, mergeClasses, shorthands } from '@griffel/react';
import { DIALOG_GAP, MEDIA_QUERY_BREAKPOINT_SELECTOR } from '../../contexts/constants';
export const dialogActionsClassNames = {
  root: 'fui-DialogActions'
};
const useStyles = /*#__PURE__*/__styles({
  root: {
    Bqenvij: "f3052tw",
    B7ck84d: "f1ewtqcl",
    mc9l5x: "f22iagw",
    Ijaq50: "f1cztu7b",
    nk6f5a: "fcu02xx",
    i8kkvl: "f4px1ci",
    Belr9w4: "fn67r4l",
    Bmdcpmo: "f6glcwc",
    th9wkt: "f1e3st1r"
  },
  gridPositionEnd: {
    Bdqf98w: "f1a7i8kp",
    Br312pm: "fd46tj4",
    Bw0ie65: "fsyjsko",
    B6n781s: "f1f41i0t"
  },
  gridPositionStart: {
    Bdqf98w: "fsxvdwy",
    Br312pm: "fwpfdsa",
    Bw0ie65: "f1e2fz10",
    Bojbm9c: "f11ihkml"
  },
  fluidStart: {
    Bw0ie65: "fsyjsko"
  },
  fluidEnd: {
    Br312pm: "fwpfdsa"
  }
}, {
  d: [".f3052tw{height:-webkit-fit-content;height:-moz-fit-content;height:fit-content;}", ".f1ewtqcl{box-sizing:border-box;}", ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".f1cztu7b{-ms-grid-row:3;grid-row-start:3;}", ".fcu02xx{-ms-grid-row-span:3;grid-row-end:3;}", ".f4px1ci{-webkit-column-gap:8px;column-gap:8px;}", ".fn67r4l{row-gap:8px;}", ".f1a7i8kp{-ms-grid-column-align:end;justify-self:end;}", ".fd46tj4{-ms-grid-column:2;grid-column-start:2;}", ".fsyjsko{-ms-grid-column-span:4;grid-column-end:4;}", ".fsxvdwy{-ms-grid-column-align:start;justify-self:start;}", ".fwpfdsa{-ms-grid-column:1;grid-column-start:1;}", ".f1e2fz10{-ms-grid-column-span:2;grid-column-end:2;}"],
  m: [["@media screen and (max-width: 480px){.f6glcwc{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}}", {
    m: "screen and (max-width: 480px)"
  }], ["@media screen and (max-width: 480px){.f1e3st1r{-ms-grid-column-align:stretch;justify-self:stretch;}}", {
    m: "screen and (max-width: 480px)"
  }], ["@media screen and (max-width: 480px){.f1f41i0t{-ms-grid-column:1;grid-column-start:1;}}", {
    m: "screen and (max-width: 480px)"
  }], ["@media screen and (max-width: 480px){.f11ihkml{-ms-grid-column-span:4;grid-column-end:4;}}", {
    m: "screen and (max-width: 480px)"
  }]]
});
/**
 * Apply styling to the DialogActions slots based on the state
 */
export const useDialogActionsStyles_unstable = state => {
  const styles = useStyles();
  state.root.className = mergeClasses(dialogActionsClassNames.root, styles.root, state.position === 'start' && styles.gridPositionStart, state.position === 'end' && styles.gridPositionEnd, state.fluid && state.position === 'start' && styles.fluidStart, state.fluid && state.position === 'end' && styles.fluidEnd, state.root.className);
  return state;
};
