// @fluentui/react-toast/lib/components/ToastContainer/renderToastContainer.js
/** @jsxRuntime classic */ /** @jsx createElement */ import { createElement } from '@fluentui/react-jsx-runtime';
import { getSlotsNext } from '@fluentui/react-utilities';
import { Transition } from 'react-transition-group';
import { ToastContainerContextProvider } from '../../contexts/toastContainerContext';
/**
 * Render the final JSX of ToastContainer
 */ export const renderToastContainer_unstable = (state, contextValues)=>{
    const { onTransitionEntering , visible , transitionTimeout , remove , nodeRef  } = state;
    const { slots , slotProps  } = getSlotsNext(state);
    return /*#__PURE__*/ createElement(Transition, {
        in: visible,
        appear: true,
        unmountOnExit: true,
        timeout: transitionTimeout,
        onExited: remove,
        onEntering: onTransitionEntering,
        nodeRef: nodeRef
    }, /*#__PURE__*/ createElement(ToastContainerContextProvider, {
        value: contextValues.toast
    }, /*#__PURE__*/ createElement(slots.root, slotProps.root), /*#__PURE__*/ createElement(slots.timer, slotProps.timer)));
};
