// @fluentui/react-drawer/lib-commonjs/components/DrawerHeaderNavigation/useDrawerHeaderNavigationStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    drawerHeaderNavigationClassNames: ()=>drawerHeaderNavigationClassNames,
    useDrawerHeaderNavigationStyles_unstable: ()=>useDrawerHeaderNavigationStyles_unstable
});
const _react = require("@griffel/react");
const drawerHeaderNavigationClassNames = {
    root: 'fui-DrawerHeaderNavigation'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        B6of3ja: "f19304h5",
        t21cq0: [
            "f1qb19o0",
            "f1p10pdi"
        ],
        jrapky: "f118i8d9",
        Frg6f3: [
            "f1p10pdi",
            "f1qb19o0"
        ],
        Bxn5svx: "f1ftwmyh"
    }
}, {
    d: [
        ".f19304h5{margin-top:calc(var(--spacingVerticalS) * -1);}",
        ".f1qb19o0{margin-right:calc(var(--spacingHorizontalL) * -1);}",
        ".f1p10pdi{margin-left:calc(var(--spacingHorizontalL) * -1);}",
        ".f118i8d9{margin-bottom:calc(var(--spacingVerticalS) * -1);}",
        ".f1ftwmyh{-webkit-order:0;-ms-flex-order:0;order:0;}"
    ]
});
const useDrawerHeaderNavigationStyles_unstable = (state)=>{
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(drawerHeaderNavigationClassNames.root, styles.root, state.root.className);
    return state;
}; //# sourceMappingURL=useDrawerHeaderNavigationStyles.styles.js.map
