// @fluentui/react-tabster/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    useArrowNavigationGroup: ()=>_index.useArrowNavigationGroup,
    useFocusableGroup: ()=>_index.useFocusableGroup,
    useFocusFinders: ()=>_index.useFocusFinders,
    useFocusVisible: ()=>_index.useFocusVisible,
    useFocusWithin: ()=>_index.useFocusWithin,
    useKeyboardNavAttribute: ()=>_index.useKeyboardNavAttribute,
    useModalAttributes: ()=>_index.useModalAttributes,
    useTabsterAttributes: ()=>_index.useTabsterAttributes,
    useObservedElement: ()=>_index.useObservedElement,
    useFocusObserved: ()=>_index.useFocusObserved,
    useMergedTabsterAttributes_unstable: ()=>_index.useMergedTabsterAttributes_unstable,
    useRestoreFocusSource: ()=>_index.useRestoreFocusSource,
    useRestoreFocusTarget: ()=>_index.useRestoreFocusTarget,
    createCustomFocusIndicatorStyle: ()=>_index1.createCustomFocusIndicatorStyle,
    createFocusOutlineStyle: ()=>_index1.createFocusOutlineStyle,
    applyFocusVisiblePolyfill: ()=>_index1.applyFocusVisiblePolyfill
});
const _index = require("./hooks/index");
const _index1 = require("./focus/index");
