// @fluentui/react-drawer/lib-commonjs/components/DrawerHeaderTitle/useDrawerHeaderTitle.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useDrawerHeaderTitle_unstable", {
    enumerable: true,
    get: ()=>useDrawerHeaderTitle_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _reactDialog = require("@fluentui/react-dialog");
const useDrawerHeaderTitle_unstable = (props, ref)=>{
    const { root: heading , action , components: titleComponents  } = (0, _reactDialog.useDialogTitle_unstable)(props, ref);
    return {
        components: {
            root: 'div',
            heading: titleComponents.root,
            action: titleComponents.action
        },
        root: (0, _reactUtilities.getNativeElementProps)('div', {
            ref,
            ...props
        }),
        heading: (0, _reactUtilities.resolveShorthand)(props.heading, {
            required: true,
            defaultProps: {
                ...heading,
                className: undefined
            }
        }),
        action: (0, _reactUtilities.resolveShorthand)(props.action, {
            defaultProps: action
        })
    };
};
