// @fluentui/react-drawer/lib-commonjs/components/DrawerInline/useDrawerInline.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useDrawerInline_unstable", {
    enumerable: true,
    get: ()=>useDrawerInline_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _getDefaultDrawerProps = require("../../util/getDefaultDrawerProps");
const useDrawerInline_unstable = (props, ref)=>{
    const { open: initialOpen , defaultOpen , size , position  } = (0, _getDefaultDrawerProps.getDefaultDrawerProps)(props);
    const { separator =false  } = props;
    const [open] = (0, _reactUtilities.useControllableState)({
        state: initialOpen,
        defaultState: defaultOpen,
        initialState: false
    });
    return {
        components: {
            root: 'div'
        },
        root: (0, _reactUtilities.getNativeElementProps)('div', {
            ref,
            ...props
        }),
        size,
        position,
        open,
        separator
    };
};
