// @fluentui/react-drawer/lib-commonjs/components/DrawerInline/useDrawerInlineStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    drawerInlineClassNames: ()=>drawerInlineClassNames,
    useDrawerInlineStyles_unstable: ()=>useDrawerInlineStyles_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _react1 = require("@griffel/react");
const _useDrawerBaseStylesStyles = require("../../util/useDrawerBaseStyles.styles");
const drawerInlineClassNames = {
    root: 'fui-DrawerInline'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react1["__styles"])({
    root: {
        qhf8xq: "f10pi13n"
    },
    separatorLeft: {
        Bekrc4i: [
            "f1hqa2wf",
            "finvdd3"
        ],
        vrafjx: [
            "fcdblym",
            "fjik90z"
        ],
        h3c5rm: [
            "fa8zu9y",
            "f17e9lqh"
        ]
    },
    separatorRight: {
        ibv6hh: [
            "finvdd3",
            "f1hqa2wf"
        ],
        wvpqe5: [
            "fjik90z",
            "fcdblym"
        ],
        zhjwy3: [
            "f17e9lqh",
            "fa8zu9y"
        ]
    }
}, {
    d: [
        ".f10pi13n{position:relative;}",
        ".f1hqa2wf{border-right-width:1px;}",
        ".finvdd3{border-left-width:1px;}",
        ".fcdblym{border-right-style:solid;}",
        ".fjik90z{border-left-style:solid;}",
        ".fa8zu9y{border-right-color:var(--colorNeutralBackground3);}",
        ".f17e9lqh{border-left-color:var(--colorNeutralBackground3);}"
    ]
});
const useDrawerInlineStyles_unstable = (state)=>{
    const baseStyles = (0, _useDrawerBaseStylesStyles.useDrawerBaseStyles)();
    const styles = useStyles();
    const separatorClass = _react.useMemo(()=>{
        if (!state.separator) {
            return undefined;
        }
        return state.position === 'left' ? styles.separatorLeft : styles.separatorRight;
    }, [
        state.position,
        state.separator,
        styles.separatorRight,
        styles.separatorLeft
    ]);
    state.root.className = (0, _react1.mergeClasses)(drawerInlineClassNames.root, baseStyles.root, styles.root, state.size && baseStyles[state.size], state.position && baseStyles[state.position], separatorClass, state.root.className);
    return state;
}; //# sourceMappingURL=useDrawerInlineStyles.styles.js.map
