// @fluentui/react-toast/lib-commonjs/components/ToastContainer/renderToastContainer.js
/** @jsxRuntime classic */ /** @jsx createElement */ "use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "renderToastContainer_unstable", {
    enumerable: true,
    get: ()=>renderToastContainer_unstable
});
const _reactJsxRuntime = require("@fluentui/react-jsx-runtime");
const _reactUtilities = require("@fluentui/react-utilities");
const _reactTransitionGroup = require("react-transition-group");
const _toastContainerContext = require("../../contexts/toastContainerContext");
const renderToastContainer_unstable = (state, contextValues)=>{
    const { onTransitionEntering , visible , transitionTimeout , remove , nodeRef  } = state;
    const { slots , slotProps  } = (0, _reactUtilities.getSlotsNext)(state);
    return /*#__PURE__*/ (0, _reactJsxRuntime.createElement)(_reactTransitionGroup.Transition, {
        in: visible,
        appear: true,
        unmountOnExit: true,
        timeout: transitionTimeout,
        onExited: remove,
        onEntering: onTransitionEntering,
        nodeRef: nodeRef
    }, /*#__PURE__*/ (0, _reactJsxRuntime.createElement)(_toastContainerContext.ToastContainerContextProvider, {
        value: contextValues.toast
    }, /*#__PURE__*/ (0, _reactJsxRuntime.createElement)(slots.root, slotProps.root), /*#__PURE__*/ (0, _reactJsxRuntime.createElement)(slots.timer, slotProps.timer)));
};
