// @fluentui/react-toast/lib-commonjs/contexts/toastContainerContext.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ToastContainerContextProvider: ()=>ToastContainerContextProvider,
    useToastContainerContext: ()=>useToastContainerContext
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const toastContainerContextDefaultValue = {
    close: ()=>null,
    intent: undefined,
    bodyId: '',
    titleId: ''
};
const ToastContainerContext = /*#__PURE__*/ _react.createContext(undefined);
const ToastContainerContextProvider = ToastContainerContext.Provider;
var _React_useContext;
const useToastContainerContext = ()=>(_React_useContext = _react.useContext(ToastContainerContext)) !== null && _React_useContext !== void 0 ? _React_useContext : toastContainerContextDefaultValue;
