// @fluentui/react-tree/lib-commonjs/components/TreeItem/useTreeItemStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    treeItemClassNames: ()=>treeItemClassNames,
    useTreeItemStyles_unstable: ()=>useTreeItemStyles_unstable
});
const _react = require("@griffel/react");
const treeItemClassNames = {
    root: 'fui-TreeItem'
};
const useRootStyles = /*#__PURE__*/ (0, _react["__styles"])({
    level1: {
        iytv0q: "f10bgyvd"
    },
    level2: {
        iytv0q: "f1h0rod3"
    },
    level3: {
        iytv0q: "fgoqafk"
    },
    level4: {
        iytv0q: "f75dvuh"
    },
    level5: {
        iytv0q: "fqk7yw6"
    },
    level6: {
        iytv0q: "f1r3z17b"
    },
    level7: {
        iytv0q: "f1hrpd1h"
    },
    level8: {
        iytv0q: "f1iy65d0"
    },
    level9: {
        iytv0q: "ftg42e5"
    },
    level10: {
        iytv0q: "fyat3t"
    },
    base: {
        qhf8xq: "f10pi13n",
        Bceei9c: "f1k6fduh",
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62",
        B7ck84d: "f1ewtqcl",
        De3pzq: "fhovq9v",
        sj55zd: "fkfq4zb",
        z189sj: [
            "flk2ux3",
            "fkl3uby"
        ]
    },
    focusIndicator: {
        Brovlpu: "ftqa4ok",
        B486eqv: "f2hkw1w",
        B8q5s1w: "f8hki3x",
        Bci5o5g: [
            "f1d2448m",
            "ffh67wi"
        ],
        n8qw10: "f1bjia2o",
        Bdrgwmp: [
            "ffh67wi",
            "f1d2448m"
        ],
        Bm4h7ae: "f15bsgw9",
        B7ys5i9: "f14e48fq",
        Busjfv9: "f18yb2kv",
        Bhk32uz: "fd6o370",
        Bf4ptjt: "fh1cnn4",
        kclons: [
            "fy7oxxb",
            "f184ne2d"
        ],
        Bhdgwq3: "fpukqih",
        Blkhhs4: [
            "f184ne2d",
            "fy7oxxb"
        ],
        Bqtpl0w: "frrh606",
        clg4pj: [
            "f1v5zibi",
            "fo2hd23"
        ],
        hgwjuy: "ful5kiu",
        Bonggc9: [
            "fo2hd23",
            "f1v5zibi"
        ],
        B1tsrr9: [
            "f1jqcqds",
            "ftffrms"
        ],
        Dah5zi: [
            "ftffrms",
            "f1jqcqds"
        ],
        Bkh64rk: [
            "f2e7qr6",
            "fsr1zz6"
        ],
        qqdqy8: [
            "fsr1zz6",
            "f2e7qr6"
        ],
        B6dhp37: "f1dvezut",
        i03rao: [
            "fd0oaoj",
            "f1cwg4i8"
        ],
        Boxcth7: "fjvm52t",
        Bsom6fd: [
            "f1cwg4i8",
            "fd0oaoj"
        ],
        J0r882: "f57olzd",
        Bule8hv: [
            "f4stah7",
            "fs1por5"
        ],
        Bjwuhne: "f480a47",
        Ghsupd: [
            "fs1por5",
            "f4stah7"
        ]
    }
}, {
    d: [
        ".f10bgyvd{--fluent-TreeItem--level:1;}",
        ".f1h0rod3{--fluent-TreeItem--level:2;}",
        ".fgoqafk{--fluent-TreeItem--level:3;}",
        ".f75dvuh{--fluent-TreeItem--level:4;}",
        ".fqk7yw6{--fluent-TreeItem--level:5;}",
        ".f1r3z17b{--fluent-TreeItem--level:6;}",
        ".f1hrpd1h{--fluent-TreeItem--level:7;}",
        ".f1iy65d0{--fluent-TreeItem--level:8;}",
        ".ftg42e5{--fluent-TreeItem--level:9;}",
        ".fyat3t{--fluent-TreeItem--level:10;}",
        ".f10pi13n{position:relative;}",
        ".f1k6fduh{cursor:pointer;}",
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".f1ewtqcl{box-sizing:border-box;}",
        ".fhovq9v{background-color:var(--colorSubtleBackground);}",
        ".fkfq4zb{color:var(--colorNeutralForeground2);}",
        ".flk2ux3{padding-right:var(--spacingHorizontalNone);}",
        ".fkl3uby{padding-left:var(--spacingHorizontalNone);}",
        ".f8hki3x[data-fui-focus-visible]{border-top-color:transparent;}",
        ".f1d2448m[data-fui-focus-visible]{border-right-color:transparent;}",
        ".ffh67wi[data-fui-focus-visible]{border-left-color:transparent;}",
        ".f1bjia2o[data-fui-focus-visible]{border-bottom-color:transparent;}",
        ".f15bsgw9[data-fui-focus-visible]::after{content:\"\";}",
        ".f14e48fq[data-fui-focus-visible]::after{position:absolute;}",
        ".f18yb2kv[data-fui-focus-visible]::after{pointer-events:none;}",
        ".fd6o370[data-fui-focus-visible]::after{z-index:1;}",
        ".fh1cnn4[data-fui-focus-visible]::after{border-top-style:solid;}",
        ".fy7oxxb[data-fui-focus-visible]::after{border-right-style:solid;}",
        ".f184ne2d[data-fui-focus-visible]::after{border-left-style:solid;}",
        ".fpukqih[data-fui-focus-visible]::after{border-bottom-style:solid;}",
        ".frrh606[data-fui-focus-visible]::after{border-top-width:2px;}",
        ".f1v5zibi[data-fui-focus-visible]::after{border-right-width:2px;}",
        ".fo2hd23[data-fui-focus-visible]::after{border-left-width:2px;}",
        ".ful5kiu[data-fui-focus-visible]::after{border-bottom-width:2px;}",
        ".f1jqcqds[data-fui-focus-visible]::after{border-bottom-right-radius:var(--borderRadiusMedium);}",
        ".ftffrms[data-fui-focus-visible]::after{border-bottom-left-radius:var(--borderRadiusMedium);}",
        ".f2e7qr6[data-fui-focus-visible]::after{border-top-right-radius:var(--borderRadiusMedium);}",
        ".fsr1zz6[data-fui-focus-visible]::after{border-top-left-radius:var(--borderRadiusMedium);}",
        ".f1dvezut[data-fui-focus-visible]::after{border-top-color:var(--colorStrokeFocus2);}",
        ".fd0oaoj[data-fui-focus-visible]::after{border-right-color:var(--colorStrokeFocus2);}",
        ".f1cwg4i8[data-fui-focus-visible]::after{border-left-color:var(--colorStrokeFocus2);}",
        ".fjvm52t[data-fui-focus-visible]::after{border-bottom-color:var(--colorStrokeFocus2);}",
        ".f57olzd[data-fui-focus-visible]::after{top:calc(2px * -1);}",
        ".f4stah7[data-fui-focus-visible]::after{right:calc(2px * -1);}",
        ".fs1por5[data-fui-focus-visible]::after{left:calc(2px * -1);}",
        ".f480a47[data-fui-focus-visible]::after{bottom:calc(2px * -1);}"
    ],
    f: [
        ".ftqa4ok:focus{outline-style:none;}"
    ],
    i: [
        ".f2hkw1w:focus-visible{outline-style:none;}"
    ]
});
const useTreeItemStyles_unstable = (state)=>{
    const rootStyles = useRootStyles();
    const { level  } = state;
    state.root.className = (0, _react.mergeClasses)(treeItemClassNames.root, isStaticallyDefinedLevel(level) && rootStyles[`level${level}`], rootStyles.base, rootStyles.focusIndicator, state.root.className);
    return state;
};
function isStaticallyDefinedLevel(level) {
    return level >= 1 && level <= 10;
} //# sourceMappingURL=useTreeItemStyles.styles.js.map
