// @fluentui/react-tree/lib-commonjs/components/TreeItemLayout/useTreeItemLayoutStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    treeItemLayoutClassNames: ()=>treeItemLayoutClassNames,
    useTreeItemLayoutStyles_unstable: ()=>useTreeItemLayoutStyles_unstable
});
const _react = require("@griffel/react");
const _treeContext = require("../../contexts/treeContext");
const _treeItemContext = require("../../contexts/treeItemContext");
const treeItemLayoutClassNames = {
    root: 'fui-TreeItemLayout',
    iconBefore: 'fui-TreeItemLayout__iconBefore',
    content: 'fui-TreeItemLayout__content',
    iconAfter: 'fui-TreeItemLayout__iconAfter',
    expandIcon: 'fui-TreeItemLayout__expandIcon',
    aside: 'fui-TreeItemLayout__aside',
    actions: 'fui-TreeItemLayout__actions',
    selector: 'fui-TreeItemLayout__selector'
};
/**
 * Styles for the root slot
 */ const useRootStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        sshi5w: "f1nxs5xn",
        B7ck84d: "f1ewtqcl",
        Ijaq50: "f15ws6j",
        Br312pm: "f135tdbu",
        nk6f5a: "f2e2169",
        Bw0ie65: "f4rqp6x",
        Bbmb7ep: [
            "f1aa9q02",
            "f16jpd5f"
        ],
        Beyfa6y: [
            "f16jpd5f",
            "f1aa9q02"
        ],
        B7oj6ja: [
            "f1jar5jt",
            "fyu767a"
        ],
        Btl43ni: [
            "fyu767a",
            "f1jar5jt"
        ],
        lj723h: "flvvhsy",
        ecr2s2: "f1wfn5kd",
        qya0sb: "f1ih54s9",
        Bi91k9c: "fnwyq0v",
        Jwef8y: "f1t94bn6",
        Becwuud: "f1jk1nfw"
    },
    leaf: {
        uwmqm3: [
            "f1k1erfc",
            "faevyjx"
        ]
    },
    branch: {
        uwmqm3: [
            "fo100m9",
            "f6yw3pu"
        ]
    },
    medium: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fkhj508",
        Bhrd7zp: "figsok6",
        Bg96gwp: "f1i3iumi"
    },
    small: {
        sshi5w: "f1pha7fy",
        Bahqtrf: "fk6fouc",
        Be2twd7: "fy9rknc",
        Bhrd7zp: "figsok6",
        Bg96gwp: "fwrc4pm"
    },
    subtle: {},
    "subtle-alpha": {
        Jwef8y: "f146ro5n",
        ecr2s2: "fkam630"
    },
    transparent: {
        De3pzq: "f1c21dwh",
        Jwef8y: "fjxutwb",
        ecr2s2: "fophhak"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".f1nxs5xn{min-height:32px;}",
        ".f1ewtqcl{box-sizing:border-box;}",
        ".f15ws6j{-ms-grid-row:layout;grid-row-start:layout;}",
        ".f135tdbu{-ms-grid-column:layout;grid-column-start:layout;}",
        ".f2e2169{-ms-grid-row-span:layout;grid-row-end:layout;}",
        ".f4rqp6x{-ms-grid-column-span:layout;grid-column-end:layout;}",
        ".f1aa9q02{border-bottom-right-radius:var(--borderRadiusMedium);}",
        ".f16jpd5f{border-bottom-left-radius:var(--borderRadiusMedium);}",
        ".f1jar5jt{border-top-right-radius:var(--borderRadiusMedium);}",
        ".fyu767a{border-top-left-radius:var(--borderRadiusMedium);}",
        ".f1k1erfc{padding-left:calc(var(--fluent-TreeItem--level, 1) * var(--spacingHorizontalXXL));}",
        ".faevyjx{padding-right:calc(var(--fluent-TreeItem--level, 1) * var(--spacingHorizontalXXL));}",
        ".fo100m9{padding-left:calc((var(--fluent-TreeItem--level, 1) - 1) * var(--spacingHorizontalXXL));}",
        ".f6yw3pu{padding-right:calc((var(--fluent-TreeItem--level, 1) - 1) * var(--spacingHorizontalXXL));}",
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fkhj508{font-size:var(--fontSizeBase300);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".f1i3iumi{line-height:var(--lineHeightBase300);}",
        ".f1pha7fy{min-height:24px;}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".fwrc4pm{line-height:var(--lineHeightBase200);}",
        ".f1c21dwh{background-color:var(--colorTransparentBackground);}"
    ],
    a: [
        ".flvvhsy:active{color:var(--colorNeutralForeground2Pressed);}",
        ".f1wfn5kd:active{background-color:var(--colorSubtleBackgroundPressed);}",
        ".f1ih54s9:active .fui-TreeItemLayout__expandIcon{color:var(--colorNeutralForeground3Pressed);}",
        ".fkam630:active{background-color:var(--colorSubtleBackgroundLightAlphaPressed);}",
        ".fophhak:active{background-color:var(--colorTransparentBackgroundPressed);}"
    ],
    h: [
        ".fnwyq0v:hover{color:var(--colorNeutralForeground2Hover);}",
        ".f1t94bn6:hover{background-color:var(--colorSubtleBackgroundHover);}",
        ".f1jk1nfw:hover .fui-TreeItemLayout__expandIcon{color:var(--colorNeutralForeground3Hover);}",
        ".f146ro5n:hover{background-color:var(--colorSubtleBackgroundLightAlphaHover);}",
        ".fjxutwb:hover{background-color:var(--colorTransparentBackgroundHover);}"
    ]
});
/**
 * Styles for the action icon slot
 */ const useActionsStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Frg6f3: [
            "fcgxt0o",
            "f1ujusj6"
        ],
        qhf8xq: "f10pi13n",
        Bj3rh1h: "f19g0ac",
        Ijaq50: "fobksn0",
        Br312pm: "fmy5l6f",
        nk6f5a: "fzqypwc",
        Bw0ie65: "f1tmftl3",
        z8tnut: "f1g0x7ka",
        z189sj: [
            "f1vdfbxk",
            "f1f5gg8d"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "f1f5gg8d",
            "f1vdfbxk"
        ]
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".fcgxt0o{margin-left:auto;}",
        ".f1ujusj6{margin-right:auto;}",
        ".f10pi13n{position:relative;}",
        ".f19g0ac{z-index:1;}",
        ".fobksn0{-ms-grid-row:aside;grid-row-start:aside;}",
        ".fmy5l6f{-ms-grid-column:aside;grid-column-start:aside;}",
        ".fzqypwc{-ms-grid-row-span:aside;grid-row-end:aside;}",
        ".f1tmftl3{-ms-grid-column-span:aside;grid-column-end:aside;}",
        ".f1g0x7ka{padding-top:0;}",
        ".f1vdfbxk{padding-right:var(--spacingHorizontalS);}",
        ".f1f5gg8d{padding-left:var(--spacingHorizontalS);}",
        ".f1qch9an{padding-bottom:0;}"
    ]
});
/**
 * Styles for the action icon slot
 */ const useAsideStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Frg6f3: [
            "fcgxt0o",
            "f1ujusj6"
        ],
        Bt984gj: "f122n59",
        Bj3rh1h: "f11zp4z2",
        Ijaq50: "fobksn0",
        Br312pm: "fmy5l6f",
        nk6f5a: "fzqypwc",
        Bw0ie65: "f1tmftl3",
        z8tnut: "f1g0x7ka",
        z189sj: [
            "fw5db7e",
            "f1uw59to"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "f1uw59to",
            "fw5db7e"
        ],
        i8kkvl: "f1ufnopg",
        Belr9w4: "f14sijuj"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".fcgxt0o{margin-left:auto;}",
        ".f1ujusj6{margin-right:auto;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".f11zp4z2{z-index:0;}",
        ".fobksn0{-ms-grid-row:aside;grid-row-start:aside;}",
        ".fmy5l6f{-ms-grid-column:aside;grid-column-start:aside;}",
        ".fzqypwc{-ms-grid-row-span:aside;grid-row-end:aside;}",
        ".f1tmftl3{-ms-grid-column-span:aside;grid-column-end:aside;}",
        ".f1g0x7ka{padding-top:0;}",
        ".fw5db7e{padding-right:var(--spacingHorizontalM);}",
        ".f1uw59to{padding-left:var(--spacingHorizontalM);}",
        ".f1qch9an{padding-bottom:0;}",
        ".f1ufnopg{-webkit-column-gap:var(--spacingHorizontalXS);column-gap:var(--spacingHorizontalXS);}",
        ".f14sijuj{row-gap:var(--spacingHorizontalXS);}"
    ]
});
/**
 * Styles for the expand icon slot
 */ const useExpandIconStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        Brf1p80: "f4d9j23",
        Bf4jedk: "f17fgpbq",
        B7ck84d: "f1ewtqcl",
        sj55zd: "f11d4kpn",
        Bh6795r: "f1jhi6b8",
        Bnnss6s: "fi64zpg",
        xawz: "f1rmlqtg",
        z8tnut: "f1ywm7hm",
        z189sj: [
            "fhxju0i",
            "f1cnd47f"
        ],
        Byoj8tv: "f14wxoun",
        uwmqm3: [
            "f1cnd47f",
            "fhxju0i"
        ]
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".f4d9j23{-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}",
        ".f17fgpbq{min-width:24px;}",
        ".f1ewtqcl{box-sizing:border-box;}",
        ".f11d4kpn{color:var(--colorNeutralForeground3);}",
        ".f1jhi6b8{-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;}",
        ".fi64zpg{-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;}",
        ".f1rmlqtg{-webkit-flex-basis:auto;-ms-flex-preferred-size:auto;flex-basis:auto;}",
        ".f1ywm7hm{padding-top:var(--spacingVerticalXS);}",
        ".fhxju0i{padding-right:0;}",
        ".f1cnd47f{padding-left:0;}",
        ".f14wxoun{padding-bottom:var(--spacingVerticalXS);}"
    ]
});
/**
 * Styles for the content slot
 */ const useContentStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        z8tnut: "f1g0x7ka",
        z189sj: [
            "ffczdla",
            "fgiv446"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "fgiv446",
            "ffczdla"
        ]
    }
}, {
    d: [
        ".f1g0x7ka{padding-top:0;}",
        ".ffczdla{padding-right:var(--spacingHorizontalXXS);}",
        ".fgiv446{padding-left:var(--spacingHorizontalXXS);}",
        ".f1qch9an{padding-bottom:0;}"
    ]
});
/**
 * Styles for the before/after icon slot
 */ const useIconStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        sj55zd: "fkfq4zb",
        Bg96gwp: "f106mvju",
        Be2twd7: "f1pp30po"
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".fkfq4zb{color:var(--colorNeutralForeground2);}",
        ".f106mvju{line-height:var(--lineHeightBase500);}",
        ".f1pp30po{font-size:var(--fontSizeBase500);}"
    ]
});
const useIconBeforeStyles = /*#__PURE__*/ (0, _react["__styles"])({
    medium: {
        z189sj: [
            "f7x41pl",
            "fruq291"
        ]
    },
    small: {
        z189sj: [
            "ffczdla",
            "fgiv446"
        ]
    }
}, {
    d: [
        ".f7x41pl{padding-right:var(--spacingHorizontalXS);}",
        ".fruq291{padding-left:var(--spacingHorizontalXS);}",
        ".ffczdla{padding-right:var(--spacingHorizontalXXS);}",
        ".fgiv446{padding-left:var(--spacingHorizontalXXS);}"
    ]
});
const useIconAfterStyles = /*#__PURE__*/ (0, _react["__styles"])({
    medium: {
        uwmqm3: [
            "fruq291",
            "f7x41pl"
        ]
    },
    small: {
        uwmqm3: [
            "fgiv446",
            "ffczdla"
        ]
    }
}, {
    d: [
        ".fruq291{padding-left:var(--spacingHorizontalXS);}",
        ".f7x41pl{padding-right:var(--spacingHorizontalXS);}",
        ".fgiv446{padding-left:var(--spacingHorizontalXXS);}",
        ".ffczdla{padding-right:var(--spacingHorizontalXXS);}"
    ]
});
const useTreeItemLayoutStyles_unstable = (state)=>{
    const { content , iconAfter , iconBefore , expandIcon , root , aside , actions , selector  } = state;
    const rootStyles = useRootStyles();
    const actionsStyles = useActionsStyles();
    const asideStyles = useAsideStyles();
    const contentStyles = useContentStyles();
    const expandIconStyles = useExpandIconStyles();
    const iconStyles = useIconStyles();
    const iconBeforeStyles = useIconBeforeStyles();
    const iconAfterStyles = useIconAfterStyles();
    const size = (0, _treeContext.useTreeContext_unstable)((ctx)=>ctx.size);
    const appearance = (0, _treeContext.useTreeContext_unstable)((ctx)=>ctx.appearance);
    const itemType = (0, _treeItemContext.useTreeItemContext_unstable)((ctx)=>ctx.itemType);
    root.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.root, rootStyles.base, rootStyles[appearance], rootStyles[size], rootStyles[itemType], root.className);
    content.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.content, contentStyles.base, content.className);
    if (expandIcon) {
        expandIcon.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.expandIcon, expandIconStyles.base, expandIcon.className);
    }
    if (iconBefore) {
        iconBefore.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.iconBefore, iconStyles.base, iconBeforeStyles[size], iconBefore.className);
    }
    if (iconAfter) {
        iconAfter.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.iconAfter, iconStyles.base, iconAfterStyles[size], iconAfter.className);
    }
    if (actions) {
        actions.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.actions, actionsStyles.base, actions.className);
    }
    if (aside) {
        aside.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.aside, asideStyles.base, aside.className);
    }
    if (expandIcon) {
        expandIcon.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.expandIcon, expandIconStyles.base, expandIcon.className);
    }
    if (selector) {
        selector.className = (0, _react.mergeClasses)(treeItemLayoutClassNames.selector, selector.className);
    }
    return state;
}; //# sourceMappingURL=useTreeItemLayoutStyles.styles.js.map
