// @fluentui/react-drawer/lib/components/DrawerHeaderNavigation/useDrawerHeaderNavigationStyles.styles.js
import { __styles, mergeClasses, shorthands } from '@griffel/react';
import { tokens } from '@fluentui/react-theme';
export const drawerHeaderNavigationClassNames = {
  root: 'fui-DrawerHeaderNavigation'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/__styles({
  root: {
    B6of3ja: "f19304h5",
    t21cq0: ["f1qb19o0", "f1p10pdi"],
    jrapky: "f118i8d9",
    Frg6f3: ["f1p10pdi", "f1qb19o0"],
    Bxn5svx: "f1ftwmyh"
  }
}, {
  d: [".f19304h5{margin-top:calc(var(--spacingVerticalS) * -1);}", ".f1qb19o0{margin-right:calc(var(--spacingHorizontalL) * -1);}", ".f1p10pdi{margin-left:calc(var(--spacingHorizontalL) * -1);}", ".f118i8d9{margin-bottom:calc(var(--spacingVerticalS) * -1);}", ".f1ftwmyh{-webkit-order:0;-ms-flex-order:0;order:0;}"]
});
/**
 * Apply styling to the DrawerHeaderNavigation slots based on the state
 */
export const useDrawerHeaderNavigationStyles_unstable = state => {
  const styles = useStyles();
  state.root.className = mergeClasses(drawerHeaderNavigationClassNames.root, styles.root, state.root.className);
  return state;
};
